### ISSUE TITLE
HidesVariable does not trigger

### ISSUE THREAD

**Issue by sebastiaanspeck** [StevenLooman/sonar-magik#289]
The next snippet should trigger HidesVariable on `_local value << _self.get_value`, since `value` was already declared in the if-statement:

```magik
_method object.current_value
	_if (value << _self.get_cached_value) _isnt _unset
	_then
		_return value
	_endif 

	_local value << _self.get_value

	_return value
_endmethod
$
```

The example above is more or less pseudo code, but unfortunately I can't share the real code.
It took my attention because loading this method generates a parser warning as well:
`[no reporter] **** Parser warning: on line 10. Variable 'value' already declared at this scope level. duplicate_variable_name`

**Comment by StevenLooman** [comment]
Fixed with https://github.com/StevenLooman/magik-tools/commit/7bb8279c21f7fc53469d57812bcf5b9a5c80eebe

**Comment by StevenLooman** [comment]
Thanks!
